feature-libs/order/components/order-details/order-detail-reorder/order-detail-reorder.component.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Component,
+  ElementRef,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { ActiveCartService } from '@spartacus/cart/base/core';
+import { CartModificationList, CartOutlets } from '@spartacus/cart/base/root';
+import { UserIdService } from '@spartacus/core';
+import { ReorderOrderFacade } from '@spartacus/order/root';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { OrderDetailsService } from '../order-details.service';
+
+@Component({
+  selector: 'cx-order-details-reorder',
+  templateUrl: './order-detail-reorder.component.html',
+})
+export class OrderDetailReorderComponent implements OnInit, OnDestroy {
+  constructor(
+    protected orderDetailsService: OrderDetailsService,
+    protected launchDialogService: LaunchDialogService,
+    protected vcr: ViewContainerRef,
+    protected userIdService: UserIdService,
+    protected reorderOrderFacade: ReorderOrderFacade,
+    protected activeCartService: ActiveCartService
+  ) {}
+
+  @ViewChild('element') element: ElementRef;
+  protected subscription = new Subscription();
+  order$: Observable<any>;
+  userId: string;
+  readonly CartOutlets = CartOutlets;
+
+  ngOnInit() {
+    this.order$ = this.orderDetailsService.getOrderDetails();
+    this.subscription.add(
+      this.userIdService.getUserId().subscribe((userId) => {
+        this.userId = userId;
+      })
+    );
+  }
+
+  onReorderClick(order: any) {
+    this.launchDialog();
+    this.reorderOrderFacade
+      .reorder(order.code, this.userId)
+      .subscribe((cartModificationList: CartModificationList) => {
+        this.activeCartService.reloadCurrentActiveCart();
+        this.launchDialogService.emitData({
+          loading: false,
+          cartModificationList,
+        });
+      });
+  }
+
```

Comment:
There seems to be another important discrepancy between the spec and the reorder behaviour.

While the feature description is all about adding the products to the cart, it appears that when doing so in the current implementation, the current active cart of the customer is deleted before adding the progucts to the cart, potentially deleting a cart full of products.

We need to be clarify if this is accepatable behaviour from the feature and if so, if we mitigate this in any way. (feedback after, warning before, save the current cart if not empty)

Reply:
Added warning message as mentioned above

Reply:
The confirmation dialog resolves this.